webservice/webservice.py

```diff
@@ -322,7 +329,79 @@ def get_manifest():
+    the S3 location if all goes well. Otherwise the HTTP status code. This that
+    can go wrong include 
+    :param response_obj: Contains the selected metadata as a TSV.
+    :type response_obj: A Flask response object.
+    :return response: Information depends on status of actions.
+    :rtype response: JSON
+    """
+    logger = logging.getLogger('dashboardService.webservice.bdbag_response')
+    # Create and return the BDbag folder.
+    bag_name = 'manifest'
+    bag_info = {'organization': '',
+                'data_type': '',
+                'date_created': datetime.datetime.now().isoformat()}
+
+    # Instantiate bag object.
+    bag = BagHandler(data=response_obj.get_data(),
+                     bag_info=bag_info,
+                     bag_name=bag_name)
+    zipped_bag = bag.create_bag()  # path to compressed bag
+    logger.info("Creating a compressed BDbag containing manifest.")
+
+    # Import bucket environment variable, launch instance of S3-file handler,
+    # and upload the BDBag file to S3.
+
+    # Transfer parameters.
+    aws_region = 'us-west-2'
+    bucket_key = str(uuid.uuid4())
+    azul_s3_bucket = os.getenv("AZUL_S3_BUCKET")
+    access_key_id = os.getenv("AWS_ACCESS_KEY_ID")
+    secret_key = os.getenv("AWS_SECRET_ACCESS_KEY")
+    url_expiration_time = 120  # in seconds
+
+    s3 = S3FileHandler(aws_region, access_key_id, secret_key)
+    r = s3.upload_object_to_bucket(azul_s3_bucket,
+                                   zipped_bag,
+                                   bucket_key)
+    os.remove(zipped_bag)
+
+    if r['status_code'] == 200:
+        logger.info("Uploaded BDbag {} to S3 bucket {}.".format(bucket_key,
+                                                                azul_s3_bucket))
+        # Generate presigned URL to access that s3 location.
+        result = s3.create_presigned_url(azul_s3_bucket, bucket_key,
+                                         url_expiration_time)
+        if result['status_code'] == 200:
+            # Happy path: bag is uploaded, and presigned URL generated.
+            logger.info("Successfully created presigned URL for bucket {}.".
+                        format(''.join([azul_s3_bucket, '/', bucket_key])))
+            response = {'url': result['presigned_url']}
+        else:
+            logger.error("Failed to create presigned URL for bucket {}.".
+                        format(''.join([azul_s3_bucket, '/', bucket_key])))
+            response = {'status': result['status_code'],
+                        'msg': 'BDBag uploaded to S3, '
+                               'but could not generate presigned URL.'}
+    else:
+        logger.error("Upload to S3 bucket {} failed: {}".format(
+            azul_s3_bucket, r['status_code']))
+        response = {
+            'msg': 'BDBag upload to S3 bucket {} failed.'
+                .format(azul_s3_bucket),
+            'status': r['status_code']
+        }
     return response
```

Comment:
You are not setting the status code of the response, which means it will always return 200 by default.

You need something like `return response, response.status` (not verified, but just to give you an idea of what I'm talking about).

Reply:
Fixed in https://github.com/DataBiosphere/azul/pull/304/commits/c26408249d016cd4a89853a439eb64059020d072.